### PR TITLE
Ci update

### PR DESCRIPTION
**Pull request by Oscar-Pepper**

### CODE REVIEW

**Comment by Oscar-Pepper** [comment]
the Dockerfile is still setup for testing our zingo-cli-in-container issue but it is cleaned up.
only added debug to source code where it's getting stuck.

File: docker-ci/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Parent image for zingolibs CI
+FROM parent-ci:latest
+
+# RUN git clone https://github.com/zingolabs/zingolib.git
+RUN git clone https://github.com/Oscar-Pepper/zingolib.git # test only
+
+# Copy zcash and lightwallet binaries into zingolib regtest directory
+WORKDIR zingolib
+RUN git checkout ci-update # test only
+RUN cp /usr/bin/lightwalletd /usr/bin/zcashd /usr/bin/zcash-cli ./regtest/bin/
```
Comment (zancas) [inline docker-ci/Dockerfile:10]: Can we use `ln -s` instead of `cp`?

Comment (zancas) [reply docker-ci/Dockerfile:10]: I suspect `ln -s` will  be more efficient/fast.

File: docker-ci/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+# Create CARGO_HOME directory for package caching
+# RUN mkdir ./cargo_home
+
+# ------------test only ----------------------
+RUN cargo build --release
```
Comment (zancas) [inline docker-ci/Dockerfile:16]: Should we add the `cross_version` feature here?

File: docker-ci/parent/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    apt install sudo -y && \
+    echo $USERNAME ALL=\(root\) NOPASSWD:ALL > /etc/sudoers.d/$USERNAME && \
+    chmod 0440 /etc/sudoers.d/$USERNAME
+USER $USERNAME
+
+# Install packages
+RUN sudo apt install curl build-essential gcc protobuf-compiler libssl-dev pkg-config git apt-transport-https wget gnupg2 unzip -y
+    
+# Install Rust
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | bash -s -- -y
```
Comment (zancas) [inline docker-ci/parent/Dockerfile:20]: Did we decide that the `-sSf` was unnecessary?

Comment (zancas) [reply docker-ci/parent/Dockerfile:20]: Nice `bash -s`

File: docker-ci/parent/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+ENV PATH="/home/${USERNAME}/.cargo/bin:${PATH}"
+
+# Install Zcash and fetch zero-knowledge parameters
+RUN wget -qO - https://apt.z.cash/zcash.asc | gpg --import && \
```
Comment (zancas) [inline docker-ci/parent/Dockerfile:24]: I think `wget` and `curl` are redundant..   it's likely you can do this with `curl` which means we don't need `wget` installed.

File: docker-ci/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# Parent image for zingolibs CI
+FROM parent-ci:latest
+
+# RUN git clone https://github.com/zingolabs/zingolib.git
```
Comment (zancas) [inline docker-ci/Dockerfile:4]: I get why you're doing this now, and I think it's the right way, but once the image is in CI, I think the version we want to test against will be available through the github CI interface.

**Comment by Oscar-Pepper** [comment]
work now tracked in new PR #219

File: docker-ci/Dockerfile (hunk 1 above)
Comment (Oscar-Pepper) [reply docker-ci/Dockerfile:10]: added in #221

File: docker-ci/Dockerfile (hunk 2 above)
Comment (Oscar-Pepper) [reply docker-ci/Dockerfile:16]: added to cargo test entrypoint in #221

File: docker-ci/Dockerfile (hunk 5 above)
Comment (Oscar-Pepper) [reply docker-ci/Dockerfile:4]: ok yes we are moving onto integrating to github now so will hook the parent image up to the CI workflow

File: docker-ci/parent/Dockerfile (hunk 3 above)
Comment (Oscar-Pepper) [reply docker-ci/parent/Dockerfile:20]: i think its fine to keep -sSf, it silences except for errors and fail fast with no output on server fails returning an error code

File: docker-ci/parent/Dockerfile (hunk 4 above)
Comment (Oscar-Pepper) [reply docker-ci/parent/Dockerfile:24]: yes nice idea, removed wget

**Pull request closed without merging**